internal/validators/boolvalidator.go

```diff
@@ -0,0 +1,116 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package validators
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-framework-validators/helpers/validatordiag"
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/schema/validator"
+	"github.com/hashicorp/terraform-plugin-framework/tfsdk"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+)
+
+// AtLeastOneOfTrueValidator is the underlying struct implementing AtLeastOneOfTrue.
```

Comment:
We can probably accomplish the same thing with some sort of combination of the built-in `any()`, `all()`, `atLeastOneOf()` bool validators with an `equals()` validator, but I think that your approach is more readable.

Reply:
Agreed. But like you, this seemed more readable to me.